### PR TITLE
fix: fixes tool metadata order and update issue, resolves agent tool metadata update failure

### PR DESCRIPTION
**Pull request by edwinjosechittilappilly**
This pull request includes several changes to the `langflow` project, focusing on deprecating the `agent_description` feature and enhancing the tools' metadata handling. The most important changes include marking the `agent_description` as deprecated, updating the tools' metadata management, and adding tags to tools.

Deprecation of `agent_description`:

* [`src/backend/base/langflow/base/agents/agent.py`](diffhunk://#diff-87ba96f91b4bbce3350ae83d5d4c7dfa3a7c800deeed21a60857dff1caaefe04L60-R64): Marked `agent_description` as deprecated and updated related comments and logic. [[1]](diffhunk://#diff-87ba96f91b4bbce3350ae83d5d4c7dfa3a7c800deeed21a60857dff1caaefe04L60-R64) [[2]](diffhunk://#diff-87ba96f91b4bbce3350ae83d5d4c7dfa3a7c800deeed21a60857dff1caaefe04L239-R247)
* Various JSON files in `src/backend/base/langflow/initial_setup/starter_projects/`: Updated the `agent_description` field to indicate deprecation. [[1]](diffhunk://#diff-5d6ded6e79298fe0d8d3e046b0fa85a89fec1f78154eb80b39acc5bdd1bb42baL1659-R1661) [[2]](diffhunk://#diff-633ea36314515f61aad893da1d70ebba0cecf77783d084343b43ca94c0e1849aL1903-R1905) [[3]](diffhunk://#diff-b589d31d615a61ea34f913c2521a443436144c496a4496b2f0ff3607b8e65d95L2150-R2152) [[4]](diffhunk://#diff-7a9ab9c72d3ef87f91707ba6d8d36aa8b2a914cc04e47d4fe6baa81b8a146e49L724-R726) [[5]](diffhunk://#diff-b26ab44cd750777d29e0f8250b23f8f5d74816b285235de6e3dfb3aa9d093fccL673-R675) [[6]](diffhunk://#diff-b26ab44cd750777d29e0f8250b23f8f5d74816b285235de6e3dfb3aa9d093fccL1251-R1253) [[7]](diffhunk://#diff-b26ab44cd750777d29e0f8250b23f8f5d74816b285235de6e3dfb3aa9d093fccL3144-R3146) [[8]](diffhunk://#diff-c85df50a21f78d9f3b397e4d15605bfe2b87804791d44651dcb3ab88881a874aL207-R209) [[9]](diffhunk://#diff-67e4d2be052585337f9b409d36f006ea24bb512085ff9a7b410e556d2567b8f2L1302-R1304) [[10]](diffhunk://#diff-67e4d2be052585337f9b409d36f006ea24bb512085ff9a7b410e556d2567b8f2L1880-R1882) [[11]](diffhunk://#diff-67e4d2be052585337f9b409d36f006ea24bb512085ff9a7b410e556d2567b8f2L2458-R2460)

Enhancements to tools' metadata handling:

* [`src/backend/base/langflow/base/tools/component_tool.py`](diffhunk://#diff-e6b607a8c5e404d2d308314cf694b33701ebcd21f8d564b6932c522d3a5e02cbR248-R278): Added a method to get tools' metadata dictionary and updated the `update_tools_metadata` method to use this dictionary.
* [`src/backend/base/langflow/custom/custom_component/component.py`](diffhunk://#diff-22fd745de33f20981c7cbcbf3c91d4a6f8719de20761eb22cf42f8ed6ec50169L1177-R1178): Updated the tools' metadata input to include tags and use the new metadata info constant. [[1]](diffhunk://#diff-22fd745de33f20981c7cbcbf3c91d4a6f8719de20761eb22cf42f8ed6ec50169L1177-R1178) [[2]](diffhunk://#diff-22fd745de33f20981c7cbcbf3c91d4a6f8719de20761eb22cf42f8ed6ec50169L1187-R1188)

Addition of tags to tools:

* [`src/backend/base/langflow/base/tools/component_tool.py`](diffhunk://#diff-e6b607a8c5e404d2d308314cf694b33701ebcd21f8d564b6932c522d3a5e02cbR221): Added tags to tools in the `get_tools` method. [[1]](diffhunk://#diff-e6b607a8c5e404d2d308314cf694b33701ebcd21f8d564b6932c522d3a5e02cbR221) [[2]](diffhunk://#diff-e6b607a8c5e404d2d308314cf694b33701ebcd21f8d564b6932c522d3a5e02cbR233)
* [`src/backend/base/langflow/base/tools/constants.py`](diffhunk://#diff-c60ffce8614e539a6fa7c3e47ac11aa7b89f15b9e09d9549c907ef15a7d33894L9-R40): Updated the tools' table schema to include tags and added a new constant for tools' metadata info.

These changes collectively improve the handling of tools' metadata and prepare for the eventual removal of the deprecated `agent_description` feature.

### CODE REVIEW

**Comment by edwinjosechittilappilly** [comment]
Merge only after: https://github.com/langflow-ai/langflow/pull/5216

**Comment by edwinjosechittilappilly** [comment]
Testing Ongoing!

File: src/backend/base/langflow/base/tools/component_tool.py [hunk 1]
```diff
@@ -243,17 +245,33 @@ def get_tools(
             raise ValueError(msg)
         return tools
 
+    def get_tools_metadata_dictionary(self) -> dict:
+        if isinstance(self.metadata, pd.DataFrame):
+            try:
+                return {record["tags"][0]: record for record in self.metadata.to_dict(orient="records")}
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/base/tools/component_tool.py:251]:
```suggestion
                return {record["tags"][0]: record for record in self.metadata.to_dict(orient="records") if record.get("tags")}
```

File: src/backend/base/langflow/services/database/service.py [hunk 2]
```diff
@@ -62,6 +62,7 @@ def __init__(self, settings_service: SettingsService):
         else:
             # Construct the path using the langflow directory.
             self.alembic_log_path = Path(langflow_dir) / alembic_log_file
+        self._logged_pragma = False
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/services/database/service.py:65]:
```suggestion
```

**Review by ogabrielluiz — APPROVED** [review]

**Pull request merged**